ingest/sqs/ingester.go

```diff
@@ -51,6 +52,15 @@ func (i *sqsIngester) ProcessAllBlockData(ctx sdk.Context) error {
 		return err
 	}
 
+	// Call the callback function for each cwPool so we can create the initial cwPool address to pool object mapping
+	// This then gets modified as cwPools are created and/or migrated
+
+	for _, cwPool := range cosmWasmPools {
+		if onCosmWasmPool != nil {
+			onCosmWasmPool(cwPool)
+		}
+	}
```

Comment:
Can we instead return the cosmwasm pools and let the caller iterate over them?

Reply:
Lol I actually had it like that but this felt even cleaner, I can revert to how I originally had it though

Reply:
While the trade-off is negligible today, I think that callbacks are more difficult to maintain and keep testable as the system evolves long-term